Source/WebCore/editing/EditorCommand.cpp

```diff
@@ -1012,20 +1012,22 @@ static bool executeRemoveFormat(LocalFrame& frame, Event*, EditorCommandSource,
     return true;
 }
 
-static bool executeScrollPageBackward(LocalFrame& frame, Event* event, EditorCommandSource, const String&)
+static bool executeScrollPageBackward(LocalFrame& frame, Event*, EditorCommandSource, const String&)
 {
     if (frame.eventHandler().shouldUseSmoothKeyboardScrollingForFocusedScrollableArea()) {
-        auto isKeyRepeat = event && event->isKeyboardEvent() && downcast<KeyboardEvent>(*event).repeat();
+        auto isKeyRepeat = frame.eventHandler().isProcessingKeyRepeatForPotentialScroll();
```

Comment:
Not really a fan of storing transient state on some class and reading it somewhere else. Can we not pass this state along the call stack in function arguments?

Reply:
I'm not either. I tried not using state and instead passing it in as arguments, but it turns out it's not possible because the stack eventually reaches into AppKit calling a selector.

Reply:
To echo @smfr's concern — is there any codepath that might result in `isProcessingKeyRepeatForPotentialScroll()` being left `true` after the key event is complete?

(If we _do_ need a transient state flag like this somewhere, I think it's also more ideal to use a RAII-like setter, or at least have the two nearby calls to `setProcessingKeyRepeatForPotentialScroll`)

Reply:
if `webpage::executeKeypressCommandsInternal` is called with a command that is not `scrollPageDown` or `scrollPageUp`, and the key pressed is a repeat, this flag will be set to `true` and not set back to `false` at the end.

However, by the time a `scrollPagee*` command is called, the flag will be reset to `false` anyways, so this is technically not the problem.

I can however make this a bit more logical and sensical by only setting the flag if the command name is `scrollPage*`.

As for RAII, this is again not possible unfortunately, because of the way the command is sent just via a selector. The stack trace is

```
eventhandler::defaultKeyboardEventHandlerWebPage::executeKeypressCommandsInternalWebPage::executeKeypressCommandsInternal commandName = scrollPageDown:WebPage::executeKeypressCommandsInternal command is NOT supportedWebViewImpl::executeSavedCommandBySelector selector = scrollPageDown:WebViewImpl::executeEditCommandForSelectorWebPageProxy::executeEditCommandEventHandler::defaultKeyboardEventHandler handleKeyboardEventWebPage::executeEditCommandEditor::Command::executeexecuteScrollPageForward
```

and so I can't think of any reasonable place a RAII scoped variable would go. If you can think of a good place for it to go, let me know!